File: 20260606/exos/04_fonction.py
# ============================================================
# Hearts of Iron — puissance de division
# ============================================================
# Une division a des régiments. La puissance totale est le nombre
# de régiments multiplié par 10.
#
# Lance : python3 04_fonction.py
# ============================================================

#   regiments : nombre de régiments dans la division
#
# puissance(3)  =>  3 * 10 = 30
# puissance(7)  =>  7 * 10 = 70
#
# Résultat attendu : puissance(3) == 30  et  puissance(7) == 70
#
# Indice : def + return + *

def puissance(regiments):
    return regiments * 10


# --- Vérification (ne pas modifier) ---
assert puissance(3) == 30, "puissance(3) doit retourner 30"
assert puissance(7) == 70, "puissance(7) doit retourner 70"
print("✅ Correct !")

File: 20260717/exos/04_fonction.lua
-- ============================================================
-- Project Zomboid — dégâts d'arme
-- ============================================================
-- Une arme inflige des dégâts de base, multipliés par un
-- modificateur (tête = 2, corps = 1). La fonction calcule le total.
--
-- Lance : lua 04_fonction.lua
-- ============================================================

--   base   : dégâts de base de l'arme
--   multi  : multiplicateur (tête, corps…)
--
-- degats(15, 2)  →  15 * 2 = 30  →  retourne 30
-- degats(20, 1)  →  20 * 1 = 20  →  retourne 20
--
-- Résultat attendu : degats(15, 2) == 30  et  degats(20, 1) == 20
--
-- Indice : function ... end  +  return

function degats(base, multi)
    total = base * multi
    return total
end


-- --- Vérification (ne pas modifier) ---
assert(degats(15, 2) == 30, "degats(15, 2) doit retourner 30")
assert(degats(20, 1) == 20, "degats(20, 1) doit retourner 20")
print("✅ Correct !")
